client/src/components/ViewAgents/index.js

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
+import React, { useState, useEffect, useContext } from 'react';
+import { Link } from "react-router-dom";
+import AuthContext from "../../utils/AuthContext";
 import classified from "../../images/classified.png";
 
 import './style.css'
 
 
 const ViewAgents = (props) => {
 
+    const [agents, setAgents] = useState([]);
+    const auth = useContext(AuthContext);
+  
+    const getAgents = () => {
+  
+      const init = {
+        headers: {
+          "Authorization": `Bearer ${auth.user.token}`
+        }
+      }
+  
+      fetch('http://localhost:8080/api/agent', init)
+        .then(response => response.json())
+        .then(data => setAgents(data))
+        .catch(error => console.log(error));
+    };
+  
+    useEffect(() => {
+      getAgents();
+    }, []);
 
     return (
 
         <main className="content">
             <div className="agentsBanner">
                 <h2>All Agents</h2>
-                <button onClick={props.addAgent} id="addAgentBtn">Add Agent +</button>
+                <Link to="/agents/add" className="link" id="addAgentBtn">Add Agent +</Link>
             </div>
+            <br/>
+            <hr/>
+            <br/>
             <div className="agentsDisplay">
-                {props.agents.map(a => (
+                {agents.map(a => (
                     <div className="agentCard" key={`${a.firstName}-p-${a.lastName}-${a.agentId}`}>
-                    <header><h4>{`${a.firstName} ${a.lastName}`}</h4></header>
+                    <header><h4>{`${a.firstName} ${a.middleName} ${a.lastName}`}</h4></header>
                     <main>
                         <img
                             src={a.image ? a.image : classified}
```

Comment:
I really like this conditional to have a picture at the ready if the user doesn't put one of their own!